torch/_meta_registrations.py

```diff
@@ -2872,6 +2872,26 @@ def nonzero_static(self, *, size: int, fill_value: int = -1):
     return self.new_empty((size, self.dim()), dtype=torch.long)
 
 
+@register_meta([aten._unsafe_masked_index])
```

Comment:
If you add these checks to the decomp, you can remove the metafunction altogether.